airbyte-integrations/connectors/source-rki-covid/source_rki_covid/source.py

```diff
@@ -0,0 +1,420 @@
+        Usually contains common params e.g. pagination size etc.
+        """
+        return {}
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        """
+        TODO: Override this method to define how a response is parsed.
+        :return an iterable containing each record in the response
+        """
+        yield response.json()
+
+
+# class that contains main source germany | full-refresh
+class Germany(RkiCovidStream):
+    """Docs: https://api.corona-zahlen.org/germany"""
+
+    primary_key = None
+
+    def path(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        return "germany/"
+
+
+# class that contains source age-groups in germany. | full-refresh
+class GermanyAgeGroups(RkiCovidStream):
+    """Docs: https://api.corona-zahlen.org/germany/age-groups"""
+
+    primary_key = None
+
+    # def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+    #     return response.json().get("data")
+
+    def path(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        return "germany/age-groups"
+
+
+# Basic incremental stream
+class IncrementalRkiCovidStream(RkiCovidStream, ABC):
+
+    state_checkpoint_interval = None
+
+    @property
+    def cursor_field(self) -> str:
+        """
+        TODO
+        Override to return the cursor field used by this stream e.g: an API entity might always use created_at as the cursor field. This is
+        usually id or date based. This field's presence tells the framework this in an incremental stream. Required for incremental.
+
+        :return str: The name of the cursor field.
+        """
+        return []
+
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
+        """
+        Override to determine the latest state after reading the latest record. This typically compared the cursor_field from the latest record and
+        the current state and picks the 'most' recent cursor. This is how a stream's state is determined. Required for incremental.
+        """
+        return {}
+
+
+# source: germany/history/cases/:days | Incremental
+class GermanyHistoryCases(IncrementalRkiCovidStream):
+    """Docs: https://api.corona-zahlen.org/germany/germany/history/cases/:days"""
+
+    primary_key = None
+
+    def __init__(self, config, **kwargs):
+        super().__init__(**kwargs)
+        self.start_date = config.get("start_date")
+
+    @property
+    def source_defined_cursor(self) -> bool:
+        return False
+
+    @property
+    def cursor_field(self) -> str:
+        return "date"
+
+    def date_to_int(self, start_date) -> int:
+        diff = datetime.now() - datetime.strptime(start_date, "%Y-%m-%d")
```

Comment:
Are you expecting to have a start_date as YYYY-mm-dd and not YYYY-mm-ddT00:00:00Z? Please update your spec.json file.

Reply:
Updated spec.json